client/components/search/index.jsx

```diff
@@ -203,6 +205,11 @@ module.exports = React.createClass( {
 		if ( event.key === 'Escape' ) {
 			this.closeSearch( event );
 		}
+
+		if ( this.props.blurOnEnter && event.which === 13 ) {
```

Comment:
Yeah, I agree we don't want this to apply to desktop because that would be really annoying. Ultimately the `renderMobile` decision in `themes-search-card` [is determined by viewport.isMobile](https://github.com/Automattic/wp-calypso/blob/fix/android-keyboard-660/shared/my-sites/themes/themes-search-card/index.jsx#L18) anyway. So if this behavior is desirable for search in general (I think it is), I'd probably just apply it to search based on `isMobile` and remove the prop. Though I agree this is a dirty bit of feature detection that could stand to be improved.